Index: st-patches/st/config.def.h
===================================================================
--- st-patches.orig/st/config.def.h
+++ st-patches/st/config.def.h
@@ -836,7 +836,7 @@ static Key key[] = {
 	{ XK_4,            Mod1Mask|ControlMask,           "\033[52;7u",  0,  0,  0},
 	{ XK_5,            ControlMask,                    "\033[53;5u",  0,  0,  0},
 	{ XK_5,            Mod1Mask|ControlMask,           "\033[53;7u",  0,  0,  0},
-	{ XK_6,            ControlMask,                    "\033[54;5u",  0,  0,  0},
+	// { XK_6,            ControlMask,                    "\033[54;5u",  0,  0,  0},
 	{ XK_6,            Mod1Mask|ControlMask,           "\033[54;7u",  0,  0,  0},
 	{ XK_7,            ControlMask,                    "\033[55;5u",  0,  0,  0},
 	{ XK_7,            Mod1Mask|ControlMask,           "\033[55;7u",  0,  0,  0},
@@ -904,7 +904,7 @@ static Key key[] = {
 	{ XK_braceright,   Mod1Mask|ControlMask,           "\033[125;7u", 0,  0,  0},
 	{ XK_braceright,   Mod1Mask|ControlMask|ShiftMask, "\033[125;8u", 0,  0,  0},
 	{ XK_braceright,   Mod1Mask|ShiftMask,             "\033[125;4u", 0,  0,  0},
-	{ XK_bracketleft,  ControlMask,                    "\033[91;5u",  0,  0,  0},
+	// { XK_bracketleft,  ControlMask,                    "\033[91;5u",  0,  0,  0},
 	{ XK_bracketleft,  ControlMask|ShiftMask,          "\033[91;6u",  0,  0,  0},
 	{ XK_bracketleft,  Mod1Mask,                       "\033[91;3u",  0,  0,  0},
 	{ XK_bracketleft,  Mod1Mask|ControlMask,           "\033[91;7u",  0,  0,  0},
